### DISCUSSION TITLE
Tips on debugging server errors

### DISCUSSION THREAD

**Discussion by svale** [craftcms/cloud#30] [Q&A]
Hi

Are there any good procedures for tracing server side error that occur on a cloud instance?

My case here is that the client reports a returning but intermittent "Internal Server Error has occurred"  error message in the CP - but without further details and I'm not sure how to trace that down. The craft console project management logs view does not seem to be the right place to look...

Thanks!

**Comment by timkelty** [comment]
`Internal Server Error`s from PHP should always end up in the environment's logs (which sounds like where you are looking).

If you send your environment details to support@craft.cloud, we can try and help you track down the error.

**Reply by timkelty** [reply]
Are they appearing full-screen, or in a little red "toast" UI from JS?

**Reply by svale** [reply]
Thanks! 

I'll have another look in the env logs, but so far I'm mostly seeing lines like this
```
REPORT RequestId: 941d25ba-637e-4b0a-bc6a-c88dc8dfdbb0	Duration: 907.31 ms	Billed Duration: 908 ms	Memory Size: 2048 MB	Max Memory Used: 214 MB	
START RequestId: 4210b315-2c01-4142-ad75-297d8b827ba5 Version: $LATEST
END RequestId: 4210b315-2c01-4142-ad75-297d8b827ba5
REPORT RequestId: 4210b315-2c01-4142-ad75-297d8b827ba5	Duration: 388.91 ms	Billed Duration: 389 ms	Memory Size: 2048 MB	Max Memory Used: 214 MB	
START RequestId: 9c8fbe55-a982-45c1-bfb3-f4254e4ebdd3 Version: $LATEST
END RequestId: 9c8fbe55-a982-45c1-bfb3-f4254e4ebdd3
REPORT RequestId: 9c8fbe55-a982-45c1-bfb3-f4254e4ebdd3	Duration: 445.15 ms	Billed Duration: 446 ms	Memory Size: 2048 MB	Max Memory Used: 214 MB	
START RequestId: 944466f2-5aa6-4589-8c38-d03855497bf7 Version: $LATEST
END RequestId: 944466f2-5aa6-4589-8c38-d03855497bf7
REPORT RequestId: 944466f2-5aa6-4589-8c38-d03855497bf7	Duration: 246.44 ms	Billed Duration: 247 ms	Memory Size: 2048 MB	Max Memory Used: 214 MB	
```
Without timestamps, hard to search through. Perhaps I'm looking at the wrong logs?

The error message, as shared with us from the client, was this toaster:
![pro-UeKEE6Xr](https://github.com/user-attachments/assets/2f2a3016-2f01-45c9-ab47-e13912f689e7)
(It is in Norwegian, but basically it says "Internal Server Error") 

I can follow up with more details to support@...

**Comment by svale** [comment]
This was resolved with help from support@craft.cloud. The underlying error was that the html purifier occasionally was timing out on the maximum execution time of 28 second. And the error was not accessible after the 2 hour retention window of the log viewer.

I've seen elsewhere that it is on the roadmap to improve the system for log files, which I think is a good idea  :)
